minikube-gradle-plugin/src/main/java/com/google/cloud/tools/minikube/MinikubeExtension.java

```diff
@@ -16,16 +16,42 @@
 
 package com.google.cloud.tools.minikube;
 
+import com.google.cloud.tools.minikube.util.CommandExecutor;
+import com.google.cloud.tools.minikube.util.MinikubeDockerEnvParser;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
 import org.gradle.api.provider.PropertyState;
 
 /** Minikube configuration extension. */
 public class MinikubeExtension {
   private final PropertyState<String> minikube;
+  private final Logger logger;
+
+  // @VisibleForTesting
+  class CommandExecutorFactory {
```

Comment:
Is this class useful outside of minikubeExtension, like in the tasks?

Reply:
We could have the `CommandExecutorFactory` be a class until `util`, and `MinikubeTask` could use that instead of calling the `CommandExecutor` constructor.

Reply:
Yeah, I mean, I realize it's the clean way to get stuff testable. I was just wondering. Otherwise the extension class itself becomes hard to follow and we really just want to limit it to configuration as much as we can.

Reply:
I pushed a commit that moves it to its own class file. Another benefit is that future classes that need to mock the `CommandExecutor` can use the `CommandExecutorFactory` as well.